common/test/acceptance/pages/lms/bookmarks.py

```diff
@@ -53,11 +54,32 @@ def breadcrumbs(self):
         breadcrumbs = self.q(css=self.BOOKMARKED_BREADCRUMBS).text
         return [breadcrumb.replace('\n', '').split('-') for breadcrumb in breadcrumbs]
 
-    def click_bookmark(self, index):
+    def click_bookmarked_unit(self, index):
         """
-        Click on bookmark at index `index`
+        Click on bookmarked unit at index `index`
 
         Arguments:
             index (int): bookmark index in the list
         """
         self.q(css=self.BOOKMARKED_ITEMS_SELECTOR).nth(index).click()
+
+    def bookmark_button_visible(self):
+        """ Check if bookmark button is visible """
+        EmptyPromise(lambda: self.q(css='.bookmark-button').visible, "Bookmark button visible").fulfill()
```

Comment:
I understand. Usually we use call `wait_for_page` from the test code itself (rather than the page object) to accomplish this. I think it would be best to stick with that pattern.